src/platforms/generic_unix/socket_driver.c

```diff
@@ -497,9 +497,18 @@ static void active_recv_callback(EventListener *listener)
     } else {
         TRACE("socket_driver: received data of len: %li\n", len);
         // {tcp, pid, binary}
-        port_ensure_available(ctx, 20 + len/(TERM_BITS/8) + 1);
+        int ensure_packet_avail;
+        int binary;
+        if (socket_data->binary == TRUE_ATOM) {
+            binary = 1;
+            ensure_packet_avail = term_binary_data_size_in_terms(len);
```

Comment:
I working on the bit syntax, I realized that we always need to add BINARY_HEADER_SIZE (2 terms/words) when ensuring enough room for binaries.  (In fact, there is one place, getting system info, where we don't add BINARY_HEADER_SIZE, but I fixed that on the branch I am working on.

I am not sure we want to always add 2, outside of `term_binary_data_size_in_terms`, but that is the way we do it now.